syncer/syncer.go

```diff
@@ -87,154 +95,176 @@ func NewBlobSyncer(
 	return bs
 }
 
-func (s *BlobSyncer) StartLoop() {
+func (b *BlockIndexer) StartConcurrentSync() {
+	var wg sync.WaitGroup
+	for i := 0; i < b.config.ConcurrencyLimit; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				if len(b.blocks) > 1000 {
+					logging.Logger.Infof("Map size:%d exceeds 1000. Pausing for a while before starting workers.", len(b.blocks))
+					time.Sleep(MapSleepTime)
+				}
+				var blockID uint64
+				b.blockHeightLock.Lock()
+				blockID = b.indexBlockHeight
+				b.indexBlockHeight++
+				b.blockHeightLock.Unlock()
+
+				for {
+					err := b.sync(blockID)
+					if err != nil {
+						logging.Logger.Error("Failed to sync: ", err)
+						time.Sleep(time.Second)
+						continue
+					}
+					break
+				}
+				logging.Logger.Infof("Successfully fetched and incremented block ID to %d, Map size:%d", blockID, len(b.blocks))
+			}
+		}()
+	}
+	wg.Wait()
+}
+
+func (b *BlockIndexer) StartLoop() {
 	go func() {
-		// nextBlockID defines the block number (BSC) or slot(ETH)
-		nextBlockID, err := s.getNextBlockNumOrSlot()
+		// nextBlockID defines the block number (BSC)
+		var nextBlockID uint64
+		var err error
+		for {
+			nextBlockID, err = b.getNextBlockNum()
+			if err == nil {
+				break
+			}
+			logging.Logger.Errorf("Failed to get next block number: %v.", err)
+			time.Sleep(time.Second * 2) // Wait for 2 seconds before retrying
+		}
 		if err != nil {
 			panic(err)
 		}
-		err = s.LoadProgressAndResume(nextBlockID)
+		b.processorBlockHeight = nextBlockID
+		b.indexBlockHeight = nextBlockID
```

Comment:
it seems indexBlockHeight is useless?

Reply:
blockID = b.indexBlockHeight
b.indexBlockHeight++

The b.indexBlockHeight is used to store the current block height being processed by the syncer. This block height refers to the most recent block number obtained from the map. It is used to reduce the time spent on each database query.